itest/lnd_estimate_route_fee_test.go

```diff
@@ -0,0 +1,226 @@
+package itest
+
+import (
+	"testing"
+
+	"github.com/btcsuite/btcd/btcutil"
+	"github.com/lightningnetwork/lnd/lnrpc"
+	"github.com/lightningnetwork/lnd/lnrpc/routerrpc"
+	"github.com/lightningnetwork/lnd/lntest"
+	"github.com/lightningnetwork/lnd/lntest/node"
+	"github.com/stretchr/testify/require"
+)
+
+type estimateRouteFeeTestCase struct {
+	// name is the name of the target test case.
+	name string
+
+	// probeInitiator is the node that will initiate the probe.
+	probeInitiator *node.HarnessNode
+
+	// destination is the node that will receive the probe.
+	destination *node.HarnessNode
+
+	// amount is the amount that will be probed for.
+	amount btcutil.Amount
+
+	// routeHints are the route hints that will be used for the probe.
+	routeHints []*lnrpc.RouteHint
+
+	// timeoutSeconds is the timeout that will be used for the probe.
+	timeoutSeconds int32
+
+	// expectedRoutingFeesMsat are the expected routing fees that will be
+	// returned by the probe.
+	expectedRoutingFeesMsat int64
+
+	// expectedFailureReasons are the expected failure reasons that will be
+	// returned by the probe.
+	expectedFailureReasons lnrpc.PaymentFailureReason
+}
+
+// testEstimateRouteFee tests that we are able to successfully route a
+// payment using multiple shards across different paths.
+func testEstimateRouteFee(ht *lntest.HarnessTest) {
+	mts := newMppTestScenario(ht)
+
+	// Set up a network with three different paths Alice <-> Bob. Channel
+	// capacities are set such that the payment can only succeed if (at
+	// least) three paths are used.
+	//
+	//              _ Eve _
+	//             /       \
+	// Alice -- Carol ---- Bob --------- Paula
+	//      \              /   (private)
+	//       \__ Dave ____/
+	//
+	req := &mppOpenChannelRequest{
+		amtAliceCarol: 235_000,
+		amtAliceDave:  135_000,
+		amtCarolBob:   135_000,
+		amtCarolEve:   135_000,
+		amtDaveBob:    135_000,
+		amtEveBob:     135_000,
+		amtBobPaula:   135_000,
```

Comment:
Which aspect of the fee estimation should random channel sizes test for?

Reply:
ah good point. It would make it clear that the outcome of the test does not depend on channel size, but perhaps the value of demonstrating that isn't worth the work to randomize them.